### PR TITLE
Base Api Client and Phalanx

### PR DESCRIPTION
**Pull request by ArturKlajnerok**
This is basic Http Api client implementation that can be shared between different specific Api Clients.
There's also Phalanx Api Client that can perform checks on usernames and uses the apiclient.

Please take a look: @pchojnacki @jcellary @Grunny

### CODE REVIEW

File: apiclient/apiclient.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		return nil, err
+	}
+
+	fullUrl := client.BaseURL.ResolveReference(endpointUrl)
+
+	request, err := http.NewRequest(method, fullUrl.String(), bytes.NewBufferString(data.Encode()))
+	if err != nil {
+		return nil, err
+	}
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
```
Comment (pchojnacki) [inline apiclient/apiclient.go]: Suggestion: this could be made more generic to support other Mime Types

**Comment by pchojnacki** [comment]
Optional suggestion,  otherwise LGTM

**Comment by Grunny** [comment]
:+1: LGTM.

File: apiclient/apiclient.go (hunk 1 above)
Comment (ArturKlajnerok) [inline apiclient/apiclient.go]: We can create an api client config for that purpose when we will have such a use-case. :+1: :cat:

**Pull request merged**